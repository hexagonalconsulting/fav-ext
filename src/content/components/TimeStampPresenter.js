import React from 'react'
import PropTypes from 'prop-types';
import customizedPulseLoader from './customizedPulseLoader'
import TimeAgo from 'timeago-react'

const TimeStampPresenter = ({timeStamp, description}) => (
  <div style={{ width: 240}}>
    {description}:{' '}
    { !!timeStamp ? <TimeAgo datetime={timeStamp} locale='en' style={{ fontWeight: 'bold' }}/> : customizedPulseLoader }
  </div>
);

TimeStampPresenter.propTypes = {
  description: PropTypes.string.isRequired,
  timeStamp: PropTypes.string,
};

export default TimeStampPresenter;
