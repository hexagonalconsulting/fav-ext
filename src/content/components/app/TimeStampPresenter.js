import React from 'react'
import customizedPulseLoader from './customizedPulseLoader'
import TimeAgo from 'timeago-react'

const TimeStampPresenter = ({timeStamp, description}) => (
  <div style={{ width: 240}}>
    {description}:{' '}
    { !!timeStamp ? <TimeAgo datetime={timeStamp} locale='en' style={{ fontWeight: 'bold' }}/> : customizedPulseLoader }
  </div>
);


export default TimeStampPresenter;
