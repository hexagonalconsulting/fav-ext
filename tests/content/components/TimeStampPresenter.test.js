import React from 'react'
import TimeStampPresenter from '../../../src/content/components/TimeStampPresenter'
// this configuration code has to run in order to configure the enzyme adapter.
import  '../../enzymeConfig'
import renderer from 'react-test-renderer';
import { render } from 'enzyme';
import timekeeper from 'timekeeper'

describe('TimeStampPresenter', () =>{

  beforeAll(() => {
    const instant = new Date('2017-12-27T01:00:00+01:00');
    return timekeeper.freeze(instant);
  });

  afterAll(() => {
    return timekeeper.reset();
  });

  describe('shows the time that has passed since the timestamp it receives', () => {

    test('10 minutes from the given timestamp', () => {
      const componentWrapper = render(
        <TimeStampPresenter
          description={'my description'}
          timeStamp={'2017-12-27T01:10:00+01:00'}
        />
      );
      expect(componentWrapper.text()).toEqual('my description: in 10 minutes')
    });

    test('1 hour from from the give timestamp', () => {
      const componentWrapper = render(
        <TimeStampPresenter
          description={'my description'}
          timeStamp={'2017-12-27T02:00:00+01:00'}
        />
      );
      expect(componentWrapper.text()).toEqual('my description: in 1 hour')
    });

    test('15 seconds from from the give timestamp', () => {
      const componentWrapper = render(
        <TimeStampPresenter
          description={'my description'}
          timeStamp={'2017-12-27T01:00:15+01:00'}
        />
      );
      expect(componentWrapper.text()).toEqual('my description: in 15 seconds')
    });

  });

  describe('when does not receive a timestamp string',  () => {

    test('still shows the description given', () => {
      const componentWrapper = render(
        <TimeStampPresenter
          description={'my description'}
          timeStamp={null}
        />
      );
      expect(componentWrapper.text()).toEqual('my description: ')
    });

    test('shows the loader', () => {
      const component = renderer.create(
        <TimeStampPresenter
          description={'my description'}
          timeStamp={null}
        />
      );
      expect(component).toMatchSnapshot()
    });

  });

});

