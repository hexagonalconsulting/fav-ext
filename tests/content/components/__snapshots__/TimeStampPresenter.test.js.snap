// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimeStampPresenter when does not receive a timestamp string shows the loader 1`] = `
<div
  style={
    Object {
      "width": 240,
    }
  }
>
  my description
  :
   
  <div
    style={
      Object {
        "display": "inline-block",
      }
    }
  >
    <div>
      <div
        className="css-msggfa"
      />
      <div
        className="css-dyhi6e"
      />
      <div
        className="css-1q065sq"
      />
    </div>
  </div>
</div>
`;
